packages/frontend-2/pages/onboarding.vue

```diff
@@ -1,70 +1,107 @@
-    <!-- 
-    Note: You might be asking yourself why do we need this route: the answer is that cloning 
-    a project is not instant, and it might take some time to get it done. We want to display
-    some sort of progress to the user in the meantime. Moreover, it makes various composables 
-    more sane to use rather than in the router navigation guards.
-  -->
-    <!-- 
-    TODO: Make this page nicer :)  
-  -->
-    <div class="w-1/5 flex flex-col space-y-2 justify-center text-center">
-      <div class="text-xs text-foreground-2">{{ status }}</div>
-      <CommonLoadingBar loading />
-      <!-- <div class="mx-auto w-20"><LogoTextWhite /></div> -->
+  <HeaderWithEmptyPage empty-header>
+    <template #header-left>
+      <HeaderLogoBlock no-link />
+    </template>
+    <template #header-right>
+      <div class="flex gap-2 items-center">
+        <FormButton
+          v-if="!isOnboardingForced"
+          class="opacity-70 hover:opacity-100 p-1"
+          size="sm"
+          color="subtle"
+          @click="setUserOnboardingComplete()"
+        >
+          Skip
+        </FormButton>
+        <FormButton color="outline" @click="() => logout({ skipRedirect: false })">
+          Sign out
+        </FormButton>
+      </div>
+    </template>
+    <div class="flex flex-col items-center justify-center p-4 max-w-lg mx-auto">
+      <h1 class="text-heading-xl text-forefround mb-2 font-normal">
+        {{ currentStage === 'join' ? 'Join your teammates' : 'Tell us about yourself' }}
+      </h1>
+      <p class="text-center text-body-sm text-foreground-2 mb-8">
+        {{
+          currentStage === 'join'
+            ? 'We found a workspace that matches your email domain'
+            : 'Your answers will help us improve'
+        }}
+      </p>
+
+      <template v-if="!loading">
+        <OnboardingJoinTeammates
+          v-if="currentStage === 'join' && discoverableWorkspaces.length > 0"
+          :workspaces="discoverableWorkspaces"
+          @next="currentStage = 'questions'"
+        />
+        <OnboardingQuestionsForm v-else />
+      </template>
+      <CommonLoadingIcon v-else size="lg" />
     </div>
-  </div>
+  </HeaderWithEmptyPage>
 </template>
+
 <script setup lang="ts">
-import { useViewerTour } from '~/lib/viewer/composables/tour'
-import {
-  useProcessOnboarding,
-  FIRST_MODEL_NAME
-} from '~~/lib/auth/composables/onboarding'
-import { homeRoute, modelRoute, projectRoute } from '~~/lib/common/helpers/route'
+import { useProcessOnboarding } from '~~/lib/auth/composables/onboarding'
+import { useAuthManager } from '~/lib/auth/composables/auth'
+import { useQuery } from '@vue/apollo-composable'
+import { graphql } from '~/lib/common/generated/gql'
+import { useActiveUser } from '~~/lib/auth/composables/activeUser'
+import { CommonLoadingIcon } from '@speckle/ui-components'
+
+const discoverableWorkspacesQuery = graphql(`
+  query DiscoverableWorkspaces {
```

Comment:
Our current pattern is to define the query in the component following the naming of the component, so PagesOnboardingDiscoverableWorkspaces_ActiveUser` or something like that, and have the query in the separate file that groups them